crates/bevy_render/src/mesh/mesh/mod.rs

```diff
@@ -620,6 +609,28 @@ impl Mesh {
     pub fn morph_target_names(&self) -> Option<&[String]> {
         self.morph_target_names.as_deref()
     }
+
+    /// Normalize joint weights so they sum to 1.
+    pub fn normalize_joint_weights(&mut self) {
+        if let Some(joints) = self.attribute_mut(Self::ATTRIBUTE_JOINT_WEIGHT) {
+            let VertexAttributeValues::Float32x4(ref mut joints) = joints else {
+                panic!("unexpected joint weight format");
+            };
+
+            for weights in joints.iter_mut() {
+                let sum: f32 = weights.iter().sum();
```

Comment:
Would it be worth verifying the weights are positive first? If someone accidentally uses `with_inserted_attribute` with a mixture of positive and negative floats, this will do weird things.

Reply:
negative weights are technically not allowed in the spec, whereas not summing to precisely 1 is allowed due to rounding issues. 

but sure, no reason not to. i arbitrarily chose to clamp negatives to zero, i guess other treatments might be better (like if all are negative, maybe you should just flip the sign?) but since we're in invalid-data territory there's probably no perfect answer. if someone wants a different treatment they can always write their own normalize function.